url/resources/urltestdata.json

```diff
@@ -8173,5 +8173,39 @@
     "input": "http://foo.0x4.",
     "base": "about:blank",
     "failure": true
+  },
+  {
+    "input": "http://foo.09..",
+    "base": "about:blank",
+    "hash": "",
+    "host": "foo.09..",
+    "hostname": "foo.09..",
+    "href":"http://foo.09../",
+    "password": "",
+    "pathname": "/",
+    "port":"",
+    "protocol": "http:",
+    "search": "",
+    "username": ""
+  },
+  {
+    "input": "http://0999999999999999999/",
+    "base": "about:blank",
+    "failure": true
+  },
+  {
+    "input": "http://foo.0x",
```

Comment:
Why should this one fail?  "0x" doesn't parse as a number, does it?

Reply:
Oops, I was wrong about this.  Nevermind.

Reply:
I think it does: https://url.spec.whatwg.org/#ipv4-number-parser step 4 puts is in base 16 mode and removes the "0x" from the input string. Then in step 6 we have the empty string so we return 0 (not failure).

This may be a separate spec bug, I dunno!